File: ruby/061-bob/bob.rb
class Bob
  # @param [String] input
  # @return [String]
  def self.hey(input)
    trimmed_input = input.gsub(/\s/, '')
    is_question = trimmed_input.match(/\?$/)
    is_yelling = trimmed_input == trimmed_input.upcase && trimmed_input.match(/[a-zA-Z]/)
    is_silence = !trimmed_input.match(/\w/)

    if is_yelling && is_question
      "Calm down, I know what I'm doing!"
    elsif is_yelling
      "Whoa, chill out!"
    elsif is_question
      "Sure."
    elsif is_silence
      "Fine. Be that way!"
    else
      "Whatever."
    end
  end
end

File: go/49-bob/bob.go
package bob

import (
	"regexp"
	"strings"
)

func Hey(remark string) string {
	isQuestion := regexp.MustCompile(`\?\s*$`).MatchString(remark)
	isYelling := regexp.MustCompile(`[a-zA-Z]`).MatchString(remark) && strings.ToUpper(remark) == remark
	isNotSilence := regexp.MustCompile(`\S`).MatchString(remark)

	if isQuestion && isYelling {
		return "Calm down, I know what I'm doing!"
	} else if isQuestion {
		return "Sure."
	} else if isYelling {
		return "Whoa, chill out!"
	} else if !isNotSilence {
		return "Fine. Be that way!"
	} else {
		return "Whatever."
	}
}
